Test failures with the latest setuptools

Tne newest setuptools normalizes names of wheels and dist-info directories and that causes 7 tests to fail here:

```
=========================== short test summary info ============================
SKIPPED [1] tests/test_sdist.py:8: could not import 'flit': No module named 'flit'
SKIPPED [1] tests/test_tagopt.py:53: Cannot compile C extensions
FAILED tests/test_bdist_wheel.py::test_unicode_record - RuntimeError: generat...
FAILED tests/test_tagopt.py::test_default_tag - AssertionError: assert 'test-...
FAILED tests/test_tagopt.py::test_build_number - AssertionError: assert 'test...
FAILED tests/test_tagopt.py::test_explicit_tag - AssertionError: assert False
FAILED tests/test_tagopt.py::test_universal_tag - AssertionError: assert False
FAILED tests/test_tagopt.py::test_universal_beats_explicit_tag - AssertionErr...
FAILED tests/test_tagopt.py::test_universal_in_setup_cfg - AssertionError: as...
FAILED tests/test_tagopt.py::test_pythontag_in_setup_cfg - AssertionError: as...
FAILED tests/test_tagopt.py::test_legacy_wheel_section_in_setup_cfg - Asserti...
=================== 9 failed, 91 passed, 2 skipped in 7.66s ====================
```

The names like `Test-1.0-py32-` should be lowercased and paths like `unicode.dist-0.1.dist-info/RECORD` should contain "_" instead of ".".

Backward-incompatible workaround:
```
sed -i "s/Test-1.0/test-1.0/" tests/test_tagopt.py
sed -i "s/unicode.dist/unicode_dist/" tests/test_bdist_wheel.py
```

--- a/tests/test_bdist_wheel.py
+++ b/tests/test_bdist_wheel.py
@@ -79,9 +79,9 @@ def test_no_scripts(wheel_paths):
 
 
 def test_unicode_record(wheel_paths):
-    path = next(path for path in wheel_paths if "unicode.dist" in path)
+    path = next(path for path in wheel_paths if "unicode_dist" in path)
     with ZipFile(path) as zf:
-        record = zf.read("unicode.dist-0.1.dist-info/RECORD")
+        record = zf.read("unicode_dist-0.1.dist-info/RECORD")
 
     assert "åäö_日本語.py".encode() in record
 

--- a/tests/test_tagopt.py
+++ b/tests/test_tagopt.py
@@ -14,7 +14,7 @@ SETUP_PY = """\
 from setuptools import setup, Extension
 
 setup(
-    name="Test",
+    name="test",
     version="1.0",
     author_email="author@example.com",
     py_modules=["test"],
@@ -63,7 +63,7 @@ def test_default_tag(temp_pkg):
     assert dist_dir.is_dir()
     wheels = list(dist_dir.iterdir())
     assert len(wheels) == 1
-    assert wheels[0].name == f"Test-1.0-py{sys.version_info[0]}-none-any.whl"
+    assert wheels[0].name == f"test-1.0-py{sys.version_info[0]}-none-any.whl"
     assert wheels[0].suffix == ".whl"
 
 
@@ -76,7 +76,7 @@ def test_build_number(temp_pkg):
     assert dist_dir.is_dir()
     wheels = list(dist_dir.iterdir())
     assert len(wheels) == 1
-    assert wheels[0].name == f"Test-1.0-1-py{sys.version_info[0]}-none-any.whl"
+    assert wheels[0].name == f"test-1.0-1-py{sys.version_info[0]}-none-any.whl"
     assert wheels[0].suffix == ".whl"
 
 
@@ -89,7 +89,7 @@ def test_explicit_tag(temp_pkg):
     assert dist_dir.is_dir()
     wheels = list(dist_dir.iterdir())
     assert len(wheels) == 1
-    assert wheels[0].name.startswith("Test-1.0-py32-")
+    assert wheels[0].name.startswith("test-1.0-py32-")
     assert wheels[0].suffix == ".whl"
 
 
@@ -101,7 +101,7 @@ def test_universal_tag(temp_pkg):
     assert dist_dir.is_dir()
     wheels = list(dist_dir.iterdir())
     assert len(wheels) == 1
-    assert wheels[0].name.startswith("Test-1.0-py2.py3-")
+    assert wheels[0].name.startswith("test-1.0-py2.py3-")
     assert wheels[0].suffix == ".whl"
 
 
@@ -114,7 +114,7 @@ def test_universal_beats_explicit_tag(temp_pkg):
     assert dist_dir.is_dir()
     wheels = list(dist_dir.iterdir())
     assert len(wheels) == 1
-    assert wheels[0].name.startswith("Test-1.0-py2.py3-")
+    assert wheels[0].name.startswith("test-1.0-py2.py3-")
     assert wheels[0].suffix == ".whl"
 
 
@@ -129,7 +129,7 @@ def test_universal_in_setup_cfg(temp_pkg):
     assert dist_dir.is_dir()
     wheels = list(dist_dir.iterdir())
     assert len(wheels) == 1
-    assert wheels[0].name.startswith("Test-1.0-py2.py3-")
+    assert wheels[0].name.startswith("test-1.0-py2.py3-")
     assert wheels[0].suffix == ".whl"
 
 
@@ -144,7 +144,7 @@ def test_pythontag_in_setup_cfg(temp_pkg):
     assert dist_dir.is_dir()
     wheels = list(dist_dir.iterdir())
     assert len(wheels) == 1
-    assert wheels[0].name.startswith("Test-1.0-py32-")
+    assert wheels[0].name.startswith("test-1.0-py32-")
     assert wheels[0].suffix == ".whl"
 
 
@@ -157,7 +157,7 @@ def test_legacy_wheel_section_in_setup_cfg(temp_pkg):
     assert dist_dir.is_dir()
     wheels = list(dist_dir.iterdir())
     assert len(wheels) == 1
-    assert wheels[0].name.startswith("Test-1.0-py2.py3-")
+    assert wheels[0].name.startswith("test-1.0-py2.py3-")
     assert wheels[0].suffix == ".whl"
 
 

--- a/tests/testdata/unicode.dist/setup.py
+++ b/tests/testdata/unicode.dist/setup.py
@@ -3,7 +3,7 @@ from __future__ import annotations
 from setuptools import setup
 
 setup(
-    name="unicode.dist",
+    name="unicode_dist",
     version="0.1",
     description="A testing distribution \N{SNOWMAN}",
     packages=["unicodedist"],

